scrapper.py

```diff
@@ -26,38 +41,143 @@ class Crawler:
+        for article in content:
+            if len(self.urls) < self.max_articles:
+                link = article.find('a')
+                href = link['href']
+                self.urls.append(href)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for url in self.seed_urls:
+            response = requests.get(url, HEADERS)  # get html code
+            sleep(random.randrange(2, 5))
+
+            article_bs = BeautifulSoup(response.text, 'html.parser')  # creates BS object
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.urls
+
+
+class HTMLParser:
+    """
+    Parser implementation
+    """
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def parse(self):
+        """
+        Extracts all necessary data from the article web page
+        """
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills the Article instance with text
+        """
+        fulltext = article_bs.find('div', {'class': 'fulltext'})
+        page_link = fulltext.find('a')['href']  # link to a page with pdf
+
+        response_pdf = requests.get(page_link, HEADERS)  # downloads page with pdf to find a link for download pdf
+        pdf_bs = BeautifulSoup(response_pdf.text, 'html.parser')
+
+        container = pdf_bs.find('div', id='pdfDownloadLinkContainer')
+        download_link = container.find('a')['href']
+
+        pdf = PDFRawFile(download_link, self.article_id)
+
+        pdf.download()
+        pdf_text = pdf.get_text()
+        if 'Список литературы' in pdf_text:
+            split_pdf = pdf_text.split('Список литературы')
+            self.article.text = split_pdf[0]
+        else:
+            self.article.text = pdf.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills the Article instance with meta information
+        """
+        article_title = article_bs.find('meta', {'name': 'description'})['content']
+        self.article.title = article_title
+
+        article_author = article_bs.find('meta', {'name': 'DC.Creator.PersonalName'})
+        if not article_author:
+            article_author = 'NOT FOUND'
+        else:
+            article_author = article_author['content']
+        self.article.author = article_author
+
+        date_raw = article_bs.find('meta', {'name': 'DC.Date.dateSubmitted'})['content']
+        article_date = datetime.datetime.strptime(date_raw, '%Y-%m-%d')
```

Comment:
what about finding date in a separate function? SRP

Reply:
If you don't mind, I will answer here on your overall comment. Can you tell me why checks do not work anymore? I don't know how to start checks, they don't do it automatically and git hub writes that there are 0 checks. And now my PR don't have any status (green or red)

Reply:
done